### PR TITLE
Lab3 angular v14 upgrades

### PR DESCRIPTION
**Pull request by suhussai**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Lab3/client/Application/README.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
-# Application
+# Dashboard
```
Comment (ujwalbukka) [inline Lab3/client/Application/README.md:1]: Application

File: Lab3/client/Application/angular.json [hunk 2]
```diff
@@ -1,165 +1,112 @@
...
               "budgets": [
+                {
+                  "type": "initial",
+                  "maximumWarning": "2mb",
+                  "maximumError": "2mb"
+                },
                 {
                   "type": "anyComponentStyle",
-                  "maximumWarning": "6kb"
+                  "maximumWarning": "2kb",
```
Comment (ujwalbukka) [inline Lab3/client/Application/angular.json:50]: 6kb

File: Lab3/client/Application/karma.conf.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ module.exports = function (config) {
       suppressAll: true // removes the duplicated traces
     },
     coverageReporter: {
-      dir: require('path').join(__dirname, './coverage/Application'),
+      dir: require('path').join(__dirname, './coverage/dashboard'),
```
Comment (ujwalbukka) [inline Lab3/client/Application/karma.conf.js:28]: Application

File: Lab3/client/Application/package.json [hunk 4]
```diff
@@ -1,73 +1,48 @@
 {
-  "name": "application",
+  "name": "dashboard",
```
Comment (ujwalbukka) [inline Lab3/client/Application/package.json:2]: application

File: Lab3/client/Application/src/app/nav/nav.component.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          e instanceof NavigationEnd ||
+          e instanceof NavigationCancel ||
+          e instanceof NavigationError
+      ),
+      map((e) => e instanceof NavigationStart)
+    );
+  }
+
+  ngOnInit(): void {
+    if (!localStorage.getItem('tenantName')) {
```
Comment (ujwalbukka) [inline Lab3/client/Application/src/app/nav/nav.component.ts:55]: If tenantName is not set you are assuming that user in not authenticated right? Can you just check this using some amplify module?

File: Lab3/client/Application/src/app/views/error/unauthorized.component.html [hunk 6]
```diff
@@ -23,14 +21,23 @@ <h1 class="display-2 text-center">Unauthorized</h1>
...
+          <mat-card-actions>
+            <div class="button-panel">
+              <button
+                mat-raised-button
+                color="primary"
+                type="submit"
+                [disabled]="!tenantForm.valid && tenantNameRequired"
+              >
+                <div *ngIf="tenantNameRequired; else loginTextBlock">Submit</div>
+                <ng-template #loginTextBlock>Login</ng-template>
```
Comment (ujwalbukka) [inline Lab3/client/Application/src/app/views/error/unauthorized.component.html:35]: Just FMI, Which "Login" template will load here

File: Lab3/client/Application/src/index.html [hunk 7]
```diff
@@ -1,13 +1,29 @@
...
+  <head>
+    <!-- https://github.com/aws/aws-amplify/issues/678 fix: -->
+    <script>
+      if (global === undefined) {
+        var global = window;
+      }
+    </script>
+    <!-- https://github.com/aws/aws-amplify/issues/678 fix end-->
+    <meta charset="utf-8" />
+    <title>Dashboard</title>
```
Comment (ujwalbukka) [inline Lab3/client/Application/src/index.html:12]: Application

File: Lab3/client/Application/src/app/nav/nav.component.ts (hunk 5 above)
Comment (suhussai) [reply Lab3/client/Application/src/app/nav/nav.component.ts:55]: At that point, we can't really assume that amplify has been configured. So, all I do is check if for the tenantName in the localStorage. If it's not set, it's a good sign that amplify has not been configured. If it does exist, then we configure Amplify and use that to check if the user is logged in.

**Pull request merged**